### PR TITLE
Fix/refresh

### PR DESCRIPTION
**Pull request by Lyncee59**
## Description
- This update contains lot of changes related to webSocket and the refresh action, in order to refresh the transactions when a new block is notified, or when the user clicks on refresh icon, by being on the transactions pages

## Change Type
Please enter one or more of the following: 
- Bug Fix

## Testing Steps
Detail the steps required for the reviewer(s) to verify and test these changes.

## Code Checklist
- [x] Code compiles successfully (verified via `yarn start`)
- [x] No lint issues exist (verified via `yarn lint`)
- [x] New and existing unit tests pass (verified via `yarn test`)
- [x] `README.md` and other documentation is updated as needed

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/actions.js [hunk 1]
```diff
@@ -24,6 +25,7 @@ export {
   goals,
   logs,
   form,
+  middleware,
```
Comment (tony-blockchain) [inline packages/blockchain-wallet-v4-frontend/src/data/actions.js]: Middleware is a very specific concept in redux, that would be a confusing name
I'd suggest websockets or network or api

File: packages/blockchain-wallet-v4-frontend/src/scenes/Home/BalancesChart/selectors.js [hunk 2]
```diff
@@ -21,12 +21,19 @@ export const getBchBalance = state => createDeepEqualSelector(
...
 
-export const getEthBalance = selectors.core.data.ethereum.getBalance
+export const getEthBalance = state => createDeepEqualSelector(
+  [
+    selectors.core.data.ethereum.getBalance
+  ],
+  (balance) => {
+    return Remote.of(balance.getOrElse(0))
+  }
+)(state)
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Home/BalancesChart/selectors.js]: We've gone back and forth on this one a few times. We need to `getOrElse(0)` for things like total balances.

File: packages/blockchain-wallet-v4-frontend/src/data/actions.js (hunk 1 above)
Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/actions.js]: We use websockets, api, and network in the core

**Comment by plondon** [comment]
@Lyncee59 anything you want to add before I merge this?

File: packages/blockchain-wallet-v4-frontend/src/scenes/Home/BalancesChart/selectors.js (hunk 2 above)
Comment (Lyncee59) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Home/BalancesChart/selectors.js]: Ok my bad !

File: packages/blockchain-wallet-v4-frontend/src/data/actions.js (hunk 1 above)
Comment (Lyncee59) [reply packages/blockchain-wallet-v4-frontend/src/data/actions.js]: I see, at this level it makes sense I think indeed to rename that to webSocket, will change that!

File: packages/blockchain-wallet-v4-frontend/src/scenes/Home/BalancesChart/selectors.js (hunk 2 above)
Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Home/BalancesChart/selectors.js]: No worries :)

**Pull request merged**